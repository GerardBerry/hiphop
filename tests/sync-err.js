"use strict"
"use hopscript"

const hh = require("hiphop");

const prg =
      <hh.module>
	<hh.outputsignal name="O"/>
	<hh.localsignal name="L">
	  <hh.parallel>

	    <hh.loop>
	      <hh.emit signal="L"/>
	      <hh.pause/>
	    </hh.loop>

	    <hh.loop>
	      <hh.await immediate signal="L"/>
	      <hh.emit signal="O"/>
	    </hh.loop>

	  </hh.parallel>
	</hh.localsignal>
      </hh.module>;

const machine = new hh.ReactiveMachine(prg, "sync-err");

try {
   console.log(machine.react());
   console.log(machine.react());
   console.log(machine.react());
   console.log(machine.react());
} catch (e) {
   console.log(e.message);
}
