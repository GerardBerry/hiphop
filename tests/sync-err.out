*** RUNTIME ERROR
*** [LOC DISABLED --- run hop -g to get location]
*** Instantaneous loop on incarnation 0.
